nuitka/optimizations/BytecodeDemotion.py

```diff
@@ -86,31 +88,40 @@ def demoteCompiledModuleToBytecode(module):
             "Demoting module %r to bytecode from %r." % (full_name, filename)
         )
 
-    source_code = module.getSourceCode()
-
-    # Second chance for plugins to modify source code just before turning it
-    # to bytecode.
-    source_code = Plugins.onFrozenModuleSourceCode(
-        module_name=full_name, is_package=False, source_code=source_code
-    )
+    module_importables_hash = getModuleImportableFilesHash(full_name)
+    if os.path.exists(_getCacheFilename(full_name, module_importables_hash, "txt")):
+        inclusion_logger.info("The module %r is already demoted" % (full_name))
+        already_demoted = True
+
+    else:
+        already_demoted = False
+        source_code = module.getSourceCode()
+
+        # Second chance for plugins to modify source code just before turning it
+        # to bytecode.
+        source_code = Plugins.onFrozenModuleSourceCode(
+            module_name=full_name, is_package=False, source_code=source_code
+        )
 
-    bytecode = compile(source_code, filename, "exec", dont_inherit=True)
+        bytecode = compile(source_code, filename, "exec", dont_inherit=True)
 
-    bytecode = Plugins.onFrozenModuleBytecode(
-        module_name=full_name, is_package=False, bytecode=bytecode
-    )
+        bytecode = Plugins.onFrozenModuleBytecode(
+            module_name=full_name, is_package=False, bytecode=bytecode
+        )
 
     uncompiled_module = makeUncompiledPythonModule(
         module_name=full_name,
         filename=filename,
-        bytecode=marshal.dumps(bytecode),
+        bytecode=marshal.dumps(bytecode) if not already_demoted else _loadBytecodeFromCache(_getCacheFilename(
+            full_name, module_importables_hash, "dat")
+        ),
         is_package=module.isCompiledPythonPackage(),
         user_provided=True,
         technical=False,
     )
 
     used_modules = module.getUsedModules()
-
+    print(used_modules)
```

Comment:
This `print` is causing the test failures. It happening on Windows is probably due to only it have a bytecode demoted module in tests scope.